gpjax/kernels.py

```diff
@@ -43,14 +46,35 @@ def params(self) -> dict:
     @params.setter
     def params(self, value):
         self._params = value
-
-
+        
 @dataclass
-class CombinationKernel:
+class _KernelSet:
     kernel_set: List[Kernel]
+        
+@dataclass
+class CombinationKernel(Kernel, _KernelSet):
     name: Optional[str] = "Combination kernel"
     combination_fn: Optional[Callable] = None
 
+    def __post_init__(self):
+        kernels = self.kernel_set
+        
+        if not all(isinstance(k, Kernel) for k in kernels):
+            raise TypeError("can only combine Kernel instances")  # pragma: no cover
+        
+        self.kernel_set: List[Kernel] = []
+        self._set_kernels(kernels)
+
+    def _set_kernels(self, kernels: Sequence[Kernel]) -> None:
+        # add kernels to a list, flattening out instances of this class therein
+        kernels_list: List[Kernel] = []
+        for k in kernels:
+            if isinstance(k, self.__class__):
+                kernels_list.extend(k.kernel_set)
+            else:
+                kernels_list.append(k)
+        self.kernel_set = kernels_list
```

Comment:
I'm guessing this is based on GPFlow's `Combination` kernel. We should acknowledge this in the docstring.

Reply:
Yes indeed this is.

Reply:
Resolved in [4c45f51](https://github.com/thomaspinder/GPJax/pull/43/commits/4c45f510b1a685cf611faaba7e67c12329c0de58).